dbeaver#38017 [CUBRID] Support PLCSQL (DBMS_OUTPUT, Procedure source...)

hwany7seo opened the pull request:
ISSUE : https://github.com/dbeaver/dbeaver/issues/38017
- Support DBMS Output
- View and modify the source code for a procedure in PL/CSQL
- Modify the name and parameters list of procedures

hwany7seo committed:
dbeaver#38017 [CUBRID] Support PLCSQL (DBMS_OUTPUT, Procedure source...)

- Support DBMS Output
- view the source code for a procedure in PL/CSQL
- Modify the name and list of procedures

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid.ui/OSGI-INF/l10n/bundle_ko.properties:
```diff
@@ -0,0 +1 @@
+page.org.jkiss.dbeaver.preferences.cubrid.main.name=\uD050\uBE0C\uB9AC\uB4DC \uC124\uC815
```
Please disable native2ascii conversion. All localization bundles must be in UTF-8 encoding.

hwany7seo replied:
I deleted this unnecessary message. The same goes for bundle.properties.

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid.ui/src/org/jkiss/dbeaver/ext/cubrid/ui/internal/CubridMessages_ko.properties:
```diff
@@ -0,0 +1,3 @@
+pref_page_cubrid_group_dbms_output = DBMS Output (\uC11C\uBC84 11.4 \uC774\uC0C1)
```
See previous comment

hwany7seo replied:
change unicode to utf-8

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid.ui/src/org/jkiss/dbeaver/ext/cubrid/ui/views/PrefPageCubrid.java:
```diff
@@ -41,15 +44,18 @@ public class PrefPageCubrid extends TargetPrefPage
-        return false;
+        DBPPreferenceStore store = dsContainer.getPreferenceStore();
+        return (store.contains(CubridConstants.PREF_DBMS_OUTPUT)
+				|| store.contains(CubridConstants.PREF_DBMS_OUTPUT_BUFFER_SIZE));
```
We use four spaces for indentation, not tabs. Please reformat your changes.

hwany7seo replied:
tab to space

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid.ui/src/org/jkiss/dbeaver/ext/cubrid/ui/views/PrefPageCubrid.java:
```diff
@@ -133,6 +147,18 @@ public void widgetSelected(final SelectionEvent e) {
+            Group miscGroup = UIUtils.createControlGroup(composite, CubridMessages.pref_page_cubrid_group_dbms_output, 2, GridData.FILL_HORIZONTAL, 0);
+            enableDbmsOutputCheck = UIUtils.createCheckbox(miscGroup, CubridMessages.pref_page_cubrid_checkbox_enable_dbms_output, "", true, 2);
+            dbmsOutputBufferSize = UIUtils.createLabelText(miscGroup, CubridMessages.pref_page_cubrid_label_buffer_size, "");
+            dbmsOutputBufferSize.addVerifyListener(new VerifyListener() {
```
Use `org.jkiss.dbeaver.ui.UIUtils#getIntegerVerifyListener` instead

hwany7seo replied:
change to getIntegerVerifyListener

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/edit/CubridProcedureManager.java:
```diff
@@ -53,4 +54,11 @@ protected void addObjectCreateActions(DBRProgressMonitor monitor, DBCExecutionCo
     }
+
+    @Override
+    protected void addObjectModifyActions(@NotNull DBRProgressMonitor monitor, @NotNull DBCExecutionContext executionContext, @NotNull List<DBEPersistAction> actionList, @NotNull ObjectChangeCommand objectChangeCommand, @NotNull Map<String, Object> options)
```
Please reformat so each parameter is located on its own line per our code style

hwany7seo replied:
each parameter is located on its own line per our code style

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridDataSource.java:
```diff
@@ -235,6 +247,21 @@ public DBSObject refreshObject(@NotNull DBRProgressMonitor monitor) throws DBExc
     }
 
+    @Override
+    protected void initializeContextState(
```
See previous comment

hwany7seo replied:
previous comment

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridDataSource.java:
```diff
@@ -321,4 +348,110 @@ protected CubridPrivilage fetchObject(
+
+    public boolean isSupportEnableDBMS() {
+        boolean ret = getContainer().getPreferenceStore().getBoolean(CubridConstants.PREF_DBMS_OUTPUT);
+        System.out.println("isSupportEnableDBMS : " + ret);
```
Please remove `System.out` logging.

hwany7seo replied:
deleted 'System.out'

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridDataSource.java:
```diff
@@ -321,4 +348,110 @@ protected CubridPrivilage fetchObject(
+        return supportDBMSOutputPLCSQL;
+    }
+
+    private boolean checkSupportDBMSOutput(
```
See previous comment about formatting

hwany7seo replied:
each parameter is located on its own line per our code style

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridProcedure.java:
```diff
@@ -143,7 +145,7 @@ public String getObjectDefinitionText(DBRProgressMonitor monitor, Map<String, Ob
                 this.source += "\nAS LANGUAGE JAVA NAME";
             } else {
-                this.source = "-- Procedure definition not available";
+                this.source = "-- JAVA Procedure definition not available";
```
```suggestion
                this.source = "-- Java procedure definition not available";

```

hwany7seo replied:
'JAVA P' to 'Java p'

ShadelessFox reviewed (changes requested):
Thanks for the contribution

hwany7seo committed:
- change unicode to utf-8
- tab to space
- change getIntegerVerifyListener
- each parameter is located on its own line per our code style

E1izabeth reviewed (dismissed):
Thanks for the contribution! I passed this PR to our QA team for testing

hwany7seo committed:
fixed conflict

hwany7seo committed:
- Fixed errors about deleted classes and changed functions.
- Fixed conflict

hwany7seo committed:
- Fixed conflict (2025-07-06)

hwany7seo committed:
- Remove unnecessary in MANIFEST.MF
- Change import order

hwany7seo committed:
Merge branch 'devel' into i_38017

hwany7seo committed:
- Fix duplicate getAdapter error in CubridDataSource. (Added by another PR.)
- Fix warning about checkstyle.
